waku/nwaku.go

```diff
@@ -1370,6 +1415,21 @@ func StartWakuNode(nodeName string, customCfg *WakuConfig) (*WakuNode, error) {
 		nodeCfg = *customCfg
 	}
 
+	if nodeCfg.TcpPort == 0 || nodeCfg.Discv5UdpPort == 0 {
+		tcpPort, udpPort, err := GetFreePortIfNeeded(0, 0)
+		if err != nil {
+			Error("Failed to allocate unique ports: %v", err)
+			tcpPort, udpPort = 0, 0
+		}
+
+		if nodeCfg.TcpPort == 0 {
+			nodeCfg.TcpPort = tcpPort
+		}
+		if nodeCfg.Discv5UdpPort == 0 {
+			nodeCfg.Discv5UdpPort = udpPort
+		}
+	}
```

Comment:
Small optimization so we don't try to allocate a port when it's not needed

```suggestion
        tcpPort, udpPort, err := GetFreePortIfNeeded(nodeCfg.TcpPort, nodeCfg.Discv5UdpPort)
        if err != nil {
	        Error("Failed to allocate unique ports: %v", err)
	        tcpPort, udpPort = 0, 0
        }
        
        if nodeCfg.TcpPort == 0 {
	        nodeCfg.TcpPort = tcpPort
        }
        if nodeCfg.Discv5UdpPort == 0 {
	        nodeCfg.Discv5UdpPort = udpPort
        }
```